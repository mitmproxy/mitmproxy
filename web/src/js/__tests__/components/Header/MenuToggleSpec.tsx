import * as React from "react";
import renderer from "react-test-renderer";
import {
    EventlogToggle,
    MenuToggle,
    OptionsToggle,
} from "../../../components/Header/MenuToggle";
import { fireEvent, render, screen, waitFor } from "../../test-utils";

import { enableFetchMocks } from "jest-fetch-mock";

enableFetchMocks();

describe("MenuToggle Component", () => {
    it("should render correctly", () => {
        const changeFn = jest.fn();
        const menuToggle = renderer.create(
            <MenuToggle onChange={changeFn} value={true}>
                <p>foo children</p>
            </MenuToggle>,
        );
        const tree = menuToggle.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

test("OptionsToggle", async () => {
    fetchMock.mockReject(new Error("backend missing"));

    const { asFragment, store } = render(
        <OptionsToggle name="anticache">toggle anticache</OptionsToggle>,
    );

    expect(asFragment()).toMatchSnapshot();
    fireEvent.click(screen.getByText("toggle anticache"));

    await waitFor(() =>
        expect(store.getState().ui.optionsEditor.anticache?.error).toContain(
            "backend missing",
        ),
    );
});

test("EventlogToggle", async () => {
    const { asFragment, store } = render(<EventlogToggle />);
    expect(asFragment()).toMatchSnapshot();

    expect(store.getState().eventLog.visible).toBeTruthy();
    fireEvent.click(screen.getByText("Display Event Log"));

    expect(store.getState().eventLog.visible).toBeFalsy();
});
