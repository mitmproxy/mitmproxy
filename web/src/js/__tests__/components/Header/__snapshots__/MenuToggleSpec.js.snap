// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventlogToggle Component should render and connect to state 1`] = `
<div
  className="menu-entry"
>
  <label>
    <input
      checked={true}
      onChange={[Function]}
      type="checkbox"
    />
    Display Event Log
  </label>
</div>
`;

exports[`MenuToggle Component should render correctly 1`] = `
<div
  className="menu-entry"
>
  <label>
    <input
      checked={true}
      onChange={[MockFunction]}
      type="checkbox"
    />
    <p>
      foo children
    </p>
  </label>
</div>
`;

exports[`SettingToggle Component should render and connect to state 1`] = `
<div
  className="menu-entry"
>
  <label>
    <input
      checked={true}
      onChange={[Function]}
      type="checkbox"
    />
    <p>
      foo children
    </p>
  </label>
</div>
`;
